commit d9b2aa07b1049ad82576934782cf8086339671f8 (refs/changes/34/2734/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-10-05T16:20:41-07:00 (2 years ago)
    
    HEAD-2367 sdc-login should not consider remote VMs that are not running

diff --git a/tools/bin/sdc-login b/tools/bin/sdc-login
index cdeccd7a..fc621117 100755
--- a/tools/bin/sdc-login
+++ b/tools/bin/sdc-login
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -148,13 +148,16 @@ fi
 #  * 'instance' is the VM uuid
 #  * 'service' is the smartdc_role
 #
-query="this.instance === '${zone}' \
-    || (this.service && this.service.match(/^${zone}/)) \
-    || (this.alias && this.alias.match(/^${zone}/))"
+query="this.server_id !== null \
+    && (this.instance === '${zone}' \
+        || (this.service && this.service.match(/^${zone}/)) \
+        || (this.alias && this.alias.match(/^${zone}/)))"
+if [[ $to_server == 0 ]]; then
+    query+=' && this.state === "running"'
+fi
 
 IFS=$'\n'
-results=$(sdcadm insts -H -s alias -j type=vm \
-    | json -c 'this.server_ip !== null' -c "${query}")
+results=$(sdcadm insts -H -s alias -j type=vm | json -c "${query}")
 length=$(json length <<< "${results}")
 if [[ ${length} -eq 0 ]]; then
     if [[ ${to_server} -eq 1 && \
